packages/react-next/src/components/OverflowSet/OverflowSet.base.tsx

```diff
@@ -1,257 +1,225 @@
-    super(props);
-
-    initializeComponentRef(this);
-    warnMutuallyExclusive(COMPONENT_NAME, props, {
-      doNotContainWithinFocusZone: 'focusZoneProps',
+const useKeytipRegistrations = (
+  persistedKeytips: { [uniqueID: string]: IKeytipProps },
+  keytipManager: KeytipManager,
+) => {
+  React.useEffect(() => {
+    Object.keys(persistedKeytips).forEach((keytipId: string) => {
+      const keytip = persistedKeytips[keytipId];
+      const uniqueID = keytipManager.register(keytip as IKeytipProps, true);
+      // Update map
+      persistedKeytips[uniqueID] = keytip;
+      delete persistedKeytips[keytipId];
     });
-  }
-
-  public render(): JSX.Element {
-    const {
-      items,
-      overflowItems,
-      className,
-      // eslint-disable-next-line deprecation/deprecation
-      focusZoneProps,
-      styles,
-      vertical,
-      // eslint-disable-next-line deprecation/deprecation
-      doNotContainWithinFocusZone,
-      role,
-      overflowSide = 'end',
-    } = this.props;
-
-    this._classNames = getClassNames(styles, { className, vertical });
-
-    let Tag;
-    let uniqueComponentProps;
-
-    if (doNotContainWithinFocusZone) {
-      Tag = 'div';
-      uniqueComponentProps = {
-        ...getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties),
-        ref: this._divContainer,
-      };
-    } else {
-      Tag = FocusZone;
-      uniqueComponentProps = {
-        ...getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties),
-        ...focusZoneProps,
-        componentRef: this._focusZone,
-        direction: vertical ? FocusZoneDirection.vertical : FocusZoneDirection.horizontal,
-      };
-    }
-
-    const showOverflow = overflowItems && overflowItems.length > 0;
-
-    return (
-      <Tag
-        role={role || 'group'}
-        aria-orientation={role === 'menubar' ? (vertical === true ? 'vertical' : 'horizontal') : undefined}
-        {...uniqueComponentProps}
-        className={this._classNames.root}
-      >
-        {overflowSide === 'start' && showOverflow && this._onRenderOverflowButtonWrapper(overflowItems!)}
-        {items && this._onRenderItems(items)}
-        {overflowSide === 'end' && showOverflow && this._onRenderOverflowButtonWrapper(overflowItems!)}
-      </Tag>
-    );
-  }
-
-  /**
-   * Sets focus to the first tabbable item in the OverflowSet.
-   * @param forceIntoFirstElement - If true, focus will be forced into the first element,
-   * even if focus is already in theOverflowSet
-   * @returns True if focus could be set to an active element, false if no operation was taken.
-   */
-  public focus(forceIntoFirstElement?: boolean): boolean {
-    let focusSucceeded = false;
-
+    return () => {
+      // Delete all persisted keytips saved
+      Object.keys(persistedKeytips).forEach((uniqueID: string) => {
+        keytipManager.unregister(persistedKeytips[uniqueID] as IKeytipProps, uniqueID, true);
+      });
+      persistedKeytips = {};
```

Comment:
This only clears *this function's reference to* `persistedKeytips`, not the original object. To clear the original object, the easiest way would be to add a line inside the loop to do `delete persistedKeytips[uniqueID]`.